sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala

```diff
@@ -70,6 +70,34 @@ class CollationStringExpressionsSuite extends QueryTest with SharedSparkSession
     })
   }
 
+  case class RepeatCollationTestCase[R](s: String, n: Int, collation: String, expectedResult: R)
+
+  test("Support Repeat string expression with Collation") {
+    // Supported collations
+    val checksRepeat = Seq(
+      RepeatCollationTestCase("Spark", 2, "UTF8_BINARY", "SparkSpark"),
```

Comment:
i see that you're just doing the same thing as the previous test but I'm not a fan of creating a class for each test

something i would recommend would be:
```
  checksRepeat.foreach { case (s, n, collation, expectedResult) =>
    ...
  }
```

but this is not a big deal, so it's up to you whichever you prefer